src/core/lib/security/security_connector/tls/tls_security_connector.h

```diff
@@ -145,10 +145,32 @@ class TlsServerSecurityConnector final : public grpc_server_security_connector {
 };
 
 // ---- Functions below are exposed for testing only -----------------------
+
+/** The |TlsFetchKeyMaterials| API ensures that |key_materials_config| has a
+ *  non-empty pem-key-cert pair list. This is done as follows:
+ *  - if |options| is equipped with a credential reload config, then this
+ *    methods uses credential reloading to populate |key_materials_config|, and
+ *    afterwards it populates |reload_status| with the status of this operation.
+ *    In particular, any data stored in |key_materials_config| is overwritten.
+ *  - if |options| has no credential reload config, then:
+ *    - if |key_materials_config| already has a non-empty pem-key-cert pair
+ *      list or is called by a client, then the method returns |GRPC_STATUS_OK|.
+ *    - if |key_materials_config| has an empty pem-key-cert pair list and is
+ *      called by a server, then the method return an error code.
```

Comment:
@yihuazhang  Is this still true when we give client the ability to disable server certificate verification? If client side doesn't verify servers' certificate, do we still mandate server to always provide a certificate?

Reply:
I will wait for this to get resolved before submitting.

Reply:
Thanks for bringing this up Zhen. 

Server does not know whether a client disables server certificate verification or not since the information is not communicated to the server. So the correct behavior is to require a server to always present a certificate and the client can choose whether or not to verify it.

Reply:
@ZhenLian Can this comment now be resolved?

Reply:
Yes sure. I will close this conversation.